plugin/api.go

```diff
@@ -272,6 +272,12 @@ type API interface {
 	// SendEphemeralPost creates an ephemeral post.
 	SendEphemeralPost(userId string, post *model.Post) *model.Post
 
+	// UpdateEphemeralPost updates an ephemeral message previously sent to the user.
+	UpdateEphemeralPost(userId string, post *model.Post) *model.Post
```

Comment:
I hesitate to add a new API method for this. Why can't [`UpdatePost()`](https://developers.mattermost.com/extend/plugins/server/reference/#API.UpdatePost) be used?

Reply:
This is a good question. We already had a `CreatePost` and a `SendEphemeralPost`, so I just followed the pattern there. Choosing between "extra/redundant" and "inconsistent" - it seemed that inconsistent would be worse here... And it certainly keeps the code of each API simpler. Yet, I am 0/5 here, can go either way.

Reply:
For me this is more of a question about "what the developers wants to do".

When the dev creates a post, he cares **what** kind of post that is. Hence it's fine to have `CreatePost` and `SendEphemeralPost`. But for update some kind of post this is different. The dev already decided what kind of post this is and just wants to change some thing in the post. Why should he care, what kind of post this is?
And for deleting a post, he "just wants to remove that post" and doesn't care about the kind.

Sure, it would be more work for the server to figure this out, but I think the complexity on the server part is worth.

Reply:
For an ephemeral post an Update is effectively a full rewrite - the server has no knowledge of the prior post, except for what's in the cookie (and even that is out of context, without material code changes), so it's completely up to the plugin to re-compose the post. If it weren't for the API compatibility issues, I'd have preferred to add a `sendAsUpdate bool` argument to `SendEphemeralPost`.

I completely see the point for `Delete` - but if we have the other 2 methods separated, then maybe it's worth having one for consistency?

@crspeller @jwilander opinions?

Reply:
Sure, the plugin/integration must provide the whole post. But why can't it use `UpdatePost()`?

Reply:
k, let me make the change and see.